shared/lib_resilience.cpp

```diff
@@ -0,0 +1,290 @@
+#include "lib_resilience.h"
+
+dispatch_resilience::dispatch_resilience(const dispatch_t &orig, size_t start_index) :
+        dispatch_t(orig),
+        connection(static_cast<CONNECTION>(m_batteryPower->connectionMode)),
+        start_outage_index(start_index){
+    inverter = nullptr;
+    if (connection == CONNECTION::DC_CONNECTED)
+        inverter = std::unique_ptr<SharedInverter>(new SharedInverter(*m_batteryPower->sharedInverter));
+    current_outage_index = start_outage_index;
+    met_loads_kw = 0;
+
+    m_batteryPower->canClipCharge = true;
```

Comment:
Also, shared.lib has missing externals when run in the test project from the #include "../../cmod_battery.h". The battstor constructor and destructor are unresolved externals on Windows and Test does not build so results cannot be viewed.

Reply:
I was thinking in the case of an outage, the SOC limits would be disregarded by the battery controller since those are usually in place to protect the battery from excessive wear during regular use but an outage would be an exceptional use with different priorities. 

We definitely need to disable Grid charging and enable PV charging and discharging of the battery for an autonomous system. Clip charging seems like it would make sense also.